locations/spiders/mcdonalds_lu.py

```diff
@@ -1,119 +1,46 @@
-                    "from_day": short_day,
-                    "to_day": short_day,
-                    "hours": hours,
-                }
-
-            index = index + 1
-
-        day_groups.append(this_day_group)
-
-        if not day_groups:
-            return None
-        opening_hours = ""
-        if len(day_groups) == 1 and day_groups[0]["hours"] in (
-            "00:00-23:59",
-            "00:00-00:00",
-        ):
-            opening_hours = "24/7"
-        else:
-            for day_group in day_groups:
-                if day_group["from_day"] == day_group["to_day"]:
-                    opening_hours += "{from_day} {hours}; ".format(**day_group)
-                else:
-                    opening_hours += "{from_day}-{to_day} {hours}; ".format(**day_group)
-            opening_hours = opening_hours[:-2]
-
-        return opening_hours
-
-    def parse_name(self, name):
-        match = re.search(r"<h1>(.*)</h1>", name)
-        return match.groups()[0].strip()
+    allowed_domains = ["mcd.lu"]
+    start_urls = ["https://mcd.lu/"]
 
     def parse_latlon(self, data):
         match = re.search(r"sll=(.*),(.*)&amp;ss", data)
-        return match.groups()[0].strip(), match.groups()[1].strip()
+        if match:
+            return match.groups()[0].strip(), match.groups()[1].strip()
+        else:
+            return None, None
 
     def parse_phone(self, phone):
-        match = re.search(r"Tel(.*)<br", phone)
-        return match.groups()[0].strip()
+        match = re.search(r'Tel <span itemprop="telephone" content="(.*)">', phone)
+        if match:
+            return match.groups()[0].strip()
+        else:
+            return None
 
     def parse_address(self, address):
         address = address[address.find("<h2>Adresse</h2>") + 16 : address.find("Tel")]
         match = re.sub("<[^<]+?>", "", address)
         return " ".join(match.split())
 
-    def parse_store(self, response):
-        data = response.text
-
-        name = self.parse_name(data)
-        address = self.parse_address(data)
-        phone = self.parse_phone(data)
-        lat, lon = self.parse_latlon(data)
-        properties = {
-            "ref": response.meta["ref"],
-            "phone": phone,
-            "lon": lon,
-            "lat": lat,
-            "name": name,
-            "addr_full": address,
-        }
-        opening_hours = self.store_hours(response)
-        if opening_hours:
-            properties["opening_hours"] = opening_hours
+    def parse(self, response, **kwargs):
+        for ref in response.xpath('//a[starts-with(@id, "snav_1_")]/@id').getall():
+            yield Request(
+                "https://mcd.lu/content.php?r={}&lang=de".format(ref.replace("snav_", "")), callback=self.parse_sd
+            )
 
-        yield Feature(**properties)
+    def post_process_item(self, item, response, ld_data, **kwargs):
+        item["website"] = None
+        item["addr_full"] = self.parse_address(response.text)
```

Comment:
Perhaps could be replaced by:

```" ".join(filter(None, map(str.strip, response.xpath('//div[@class="txt"]/h2/following::text()').getall()))).split(" Tel ", 1)[0]```

Reply:
Tested and agreed. It is a little easier to read.